main.nf

```diff
@@ -55,24 +55,22 @@ workflow {
         .flatMap { samplesheet ->
             samplesheetToList(samplesheet.toString(), "${projectDir}/assets/schema_input.json")
         }
-        .map { meta, fastq_1, fastq_2, alignment ->
-            def read_group = "@RG\\tID:${fastq_1.simpleName}_${meta.laneId}\\tPL:${meta.sequencer}\\tSM:${meta.id}"
-            def single_end = !fastq_2
-            def fastqs = single_end ? [fastq_1] : [fastq_1, fastq_2]
+        .map { meta, reads1, reads2, alignment ->
+            def id = meta.runId ? "${meta.sample}_${meta.runId}" : "${meta.sample}"
```

Comment:
what I understood, runId is mandatory when you use submission path, is this part only required for samplesheet inputs?

Reply:
Exactly, we currently assume the alignment input (through samplesheet) is [already merged](https://github.com/BfArM-MVH/GRZ_QC_Workflow/blob/main/subworkflows/local/fastq_align_bwa_markduplicates/main.nf#L62) by sample, so it won't have a runId.

Reply:
The logic is the same for samplesheets and submissions that run from reads. The logic is only different for alignment input, which is only possible through the samplesheet.